reply.lua

```diff
@@ -0,0 +1,166 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Ring Zero Desenvolvimento de Software LTDA
+-- SPDX-License-Identifier: GPL-2.0-only
+--
+
+local data   = require("data")
+local linux  = require("linux")
+local socket = require("socket")
+local pack   = require("dome/pack")
+local config = require("dome/config")
+
+local PACKET = socket.af.PACKET
+local RAW    = socket.sock.RAW
+
+local ETH_HLEN  = 14
+local ETH_P_ALL = 0x0003
+
+local IPPROTO_TCP = 0x06
+
+local RST = 1 << 2
+local ACK = 1 << 4
+
+local packer   = pack.packer
+local unpacker = pack.unpacker
+
+local reply = {}
+
+local ntoh16 = linux.ntoh16
+local function checksum(data, base, length)
```

Comment:
Instead of passing `data` and `base` we can pass the closures returned by `unpacker` (that's, `u8` and `be16`), right?

Reply:
what would we gain with that?

I find `checksum(data, base, length)` a lot clearer than `checksum(data, u8, be16, length)`, no? It states right upfront what it's doing.

what do you think?

Reply:
Alternatively, we could call unpacker inside checksum.. perhaps it's a better approach.. but it's not mandatory.. we can leave it as is.. up to you (whatever looks better ;-)